context.go

```diff
@@ -49,6 +50,21 @@ func (ctx Context) Get(key interface{}, defaultValue ...interface{}) (interface{
 	return ctx.values[key], nil
 }
 
+// GetAs copies data from tke key to the dest.
+// Supports maps, slices and structs.
+func (ctx Context) GetAs(key interface{}, dest interface{}) error {
+	if _, ok := ctx.values[key]; !ok {
+		return fmt.Errorf("the key %+v does not exist", key)
+	}
+
+	d, err := json.Marshal(ctx.values[key])
```

Comment:
How about using [mapstructure](https://github.com/mitchellh/mapstructure)?

Reply:
I assumed that the value can be anything and I want to keep everything as it was originally. It can be a string, int, struct, map, or anything else. So, if you set a struct of a specific type to the context, you should be able to get it back as well. The mapstructure is an additional dependency and I don't see any benefits of using it over the JSON package from the std lib.